File: 17. Letter Combinations of a Phone Number/LetterCombination.java
class Solution {
    public String[] table = { "0", "1", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz" };

    public List<String> letterCombinations(String digits) {
        LinkedList<String> output = new LinkedList<>();

        if (digits == null || digits.length() == 0) {
            return output;
        }

        output.add("");
        while (output.peek().length() != digits.length()) {
            String cur = output.poll();
            //In ASCII '0' = 48, 1='49' etc.
            int num = digits.charAt(cur.length()) - '0';
            for (char c : table[num].toCharArray()) {
                output.add(cur + c);
            }
        }
        return output;
    }
}

File: 17. Letter Combinations of a Phone Number/LetterCombination.py
class Solution:
    def letterCombinations(self, digits: str) -> List[str]:
        if not digits:
            return []
        
        table = [[], [], ['a', 'b', 'c'], ['d', 'e', 'f'], ['g', 'h', 'i'],
                 ['j', 'k', 'l'], ['m', 'n', 'o'], ['p', 'q', 'r', 's'],
                 ['t', 'u', 'v'], ['w', 'x', 'y', 'z']]
        
        output = [""]
        while output[0].length < len(digits):
            cur = output.pop(0)
            num = int(digits[len(cur)])
            for c in table[num]:
                output.append(cur + c)
        
        return output